backend/app/auth/service.py

```
@@ -127,17 +137,31 @@ async def authenticate_user_with_email(self, email: str, password: str) -> Dict[
+            
         # Create new refresh token
-        refresh_token = await self._create_refresh_token_record(str(user["_id"]))
-        
+        try:
+            refresh_token = await self._create_refresh_token_record(str(user["_id"]))
+        except Exception as e:
+            logger.error(f"Failed to generate refresh token: {e}")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Failed to generate refresh token"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)